return_type.md

```diff
@@ -0,0 +1,94 @@
+    return_type(f, T::Type{<:Tuple}) -> R::Type
```

Comment:
We could also rename this to `upper_bound_return_type`. If I was new to julia, I would have a clear (and wrong) intuition what `return_types` does and might not look it up in the manual before using it. A function with this more complicated name I would definitely look up.

Reply:
`estimate_return_type` would be another possible name.

Reply:
I prefer `upper_bound_return_type` as it clarifies the "direction" of the bound compared to `estimate_return_type`. But let's see how other people think about it.